chalice/deploy/deployer.py

```diff
@@ -59,6 +59,7 @@ def validate_configuration(config):
     """
     routes = config.chalice_app.routes
     validate_routes(routes)
+    validate_route_content_types(routes, config.chalice_app.api.binary_types)
```

Comment:
There's not really a strong reason.  I think a reasonable refactoring in the future would be to move all of this over to a validation class with each of these being subclasses.

Initially there was only a handful of validations.  There was a prior discussion above about moving more validation that can be done at deploy time out of the runtime critical path.